plugin/core/sessions.py

```diff
@@ -1253,10 +1269,16 @@ def m_client_registerCapability(self, params: Any, request_id: Any) -> None:
                     # Inform only after the response is sent, otherwise we might start doing requests for capabilities
                     # which are technically not yet done registering.
                     sublime.set_timeout_async(inform)
-            if self._plugin:
-                inform = functools.partial(
-                    self._plugin.on_register_capability_async, registration_id, capability_path, options)
-                sublime.set_timeout_async(inform)
+            if self._watcher_impl and capability_path == "didChangeWatchedFilesProvider":
```

Comment:
`didChangeWatchedFilesProvider` is this string correct?

I think that it should be `didChangeWatchedFiles`

Reply:
Our `method_to_capability` transforms the original `workspace/didChangeWatchedFiles` capability into `didChangeWatchedFilesProvider`. It's not a capability that is defined by the spec but that shouldn't really be a problem.

One benefit how this works is that the user can disable this capability using the `disabled_capabilities` option as long as he/she knows to use that specific string.

Reply:
Thanks :+1: